### PR TITLE
Support timeout and memory size in run configuration.

### PR DESCRIPTION
**Pull request by zhangzhx**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Description
* Refactor the Slider + TextField panel to a reusable component
* Rename the `LambdaRemoteRunSettingsEditorPanel` to `RemoteLambdaRunSettingsEditorPanel` to be consistent with local.
* Add `timeout` and `memorySize` to the `Function` resource when parsing CloudFormation template
* Add `timeout` and `memorySize` widgets in `LocalLambdaRunSettingsEditorPanel`

## Related Issue(s)
#976  #977 

## Screenshots (if appropriate)

Screen Shot 2019-08-01 at 11 37 27 AM
Screen Shot 2019-08-01 at 11 38 03 AM

## Checklist

- [x] I have read the **README** document
- [x] I have read the **CONTRIBUTING** document
- [x] Local run of `gradlew check` succeeds
- [x] My code follows the code style of this project
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed
- [x] A short description of the change has been added to the **CHANGELOG**

## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/sam/SamTemplateUtils.kt [hunk 1]
```diff
@@ -64,7 +66,8 @@ object SamTemplateUtils {
                         keyValue("Handler", handler)
                         keyValue("CodeUri", codeUri)
                         keyValue("Runtime", runtime.toString())
-                        keyValue("Timeout", "900")
+                        keyValue("Timeout", timeout.toString())
```
Comment (kiiadi) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/sam/SamTemplateUtils.kt]: how does this work if we hit a break-point while debugging?

File: jetbrains-core/src/software/aws/toolkits/jetbrains/ui/SliderPanel.java [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+// A panel with a slider and text field, of which the slider and text field always synced up.
+public class SliderPanel {
+    private final int min, max, minTick, maxTick, minorTick, majorTick, defaultValue;
+    private final boolean snap;
+
+    private JPanel content;
+    public JSlider slider;
+    public JTextField textField;
+
+    public static SliderPanel lambdaTimeout() {
```
Comment (kiiadi) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/ui/SliderPanel.java]: This definitely doesn't live here - the `jetbrains.ui` stuff should be pretty much service agnostic

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/sam/SamTemplateUtils.kt (hunk 1 above)
Comment (zhangzhx) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/sam/SamTemplateUtils.kt]: I've manually tested running in debug mode by calling `context.getRemainingTimeInMillis()`. Sam local set the timeout to be 10 hours ignoring the actual timeout in the template. So it should be ok in debug mode.

**Comment by zhangzhx** [comment]
> Does this only occur when _not_ running locally from a template?

Yes. When running from template, we disable the timeout and memory size settings same way as handler and runtime.

File: jetbrains-core/src/software/aws/toolkits/jetbrains/ui/SliderPanel.java (hunk 2 above)
Comment (zhangzhx) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/ui/SliderPanel.java]: Moved this to `Lambda` file.

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/sam/SamTemplateUtils.kt (hunk 1 above)
Comment (kiiadi) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/sam/SamTemplateUtils.kt]: Does it only do this in debug mode? If it does it all the time what's the point of setting the value at all?

Comment (zhangzhx) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/sam/SamTemplateUtils.kt]: It only does it in debug mode. I am waiting for the confirmation from the sam team.

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/Lambda.kt [hunk 3]
```diff
@@ -22,6 +31,25 @@ object Lambda {
...
+    const val MAX_MEMORY = 3008
+    const val MEMORY_INCREMENT = 64
+    const val DEFAULT_MEMORY_SIZE = 128
+    const val MIN_TIMEOUT = 1
+    @JvmField val MAX_TIMEOUT = TimeUnit.MINUTES.toSeconds(15).toInt()
+    @JvmField val DEFAULT_TIMEOUT = TimeUnit.MINUTES.toSeconds(5).toInt()
+}
+
+object LambdaWidgets {
+    fun lambdaTimeout(): SliderPanel =
```
Comment (kiiadi) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/Lambda.kt]: If you annotate these methods with `@JvmStatic` you won't have to grab the `INSTANCE` in the java code they'll appear as top-level static methods.

Comment (zhangzhx) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/Lambda.kt]: Done.

**Pull request merged**